pretyping/libBinding.ml

```diff
@@ -80,23 +80,23 @@ struct
   let add_names names decl =
     match get_name decl with
     | Anonymous -> names
-    | Name id -> Id.Set.add id names
+    | Name id -> Nameops.Fresh.add id names
 
   let fresh_key s = List.length (state_context s)
 
   let push_old_rel decl s =
   let s' = { s with
       env = EConstr.push_rel decl s.env ;
       names = add_names s.names decl;
-      subst = mkRel 1 :: List.map (Vars.lift 1) s.subst
+      subst = Esubst.el_lift s.subst;
     } in
   (s', fresh_key s)
 
   let push_fresh_rel decl s =
     let s' = { s with
       env = EConstr.push_rel decl s.env ;
       names = add_names s.names decl;
-      subst = List.map (Vars.lift 1) s.subst
+      subst = Esubst.el_shft 1 (Esubst.el_lift s.subst);
```

Comment:
Should we create a new `el_weak` operation for this ? Because it looks like it should really be a primitive for lifts